### PR TITLE
Mcp 1894 - Fix a bug and add unit test code coverage in service-data-access.

### PR DESCRIPTION
**Pull request by wlin2019**
Fixing a bug and adding unit tests to increase coverage in service-data-access

## Description

Added unit tests to increase the code coverage to 80%. While adding the unit tests, a bug was detected. The bug fix is included in this PR. Also updated jwt package versions (see PR#379, #380).

### What was the problem?

The service-data-access unit test code coverage was around 65%. Added the unit tests to increase the code coverage to 80%. The added unit tests detected a bug in FieldExtractor for handling DateTimeType value, and it got fixed in this PR.

### How does this fix it?

See MCP-1894. The attachments in the Jira ticket show the code coverage increase.

### Jira Tickets

MCP-1894

- [MCP-000](https://amida.atlassian.net/browse/MCP-000)

## How to test this PR

- Step 1. Build the application.
- Step 2. Go to service-data-access Jacobo page to check the code coverage. It should show 80% as shown in the attached screen shot in MCP-1894.

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
- [ ] If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) and
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
- [ ] If changing software behavior, post a link to the PR in Slack channel #benefits-virtual-regional-office

### CODE REVIEW

File: service-data-access/src/test/java/gov/va/vro/abd_data_access/service/FhirClientTest.java [hunk 1]
```diff
@@ -4,10 +4,7 @@
 import ca.uhn.fhir.parser.IParser;
 import ca.uhn.fhir.rest.api.EncodingEnum;
 import gov.va.vro.abd_data_access.exception.AbdException;
-import gov.va.vro.abd_data_access.model.AbdBloodPressure;
-import gov.va.vro.abd_data_access.model.AbdClaim;
-import gov.va.vro.abd_data_access.model.AbdEvidence;
-import gov.va.vro.abd_data_access.model.AbdMedication;
+import gov.va.vro.abd_data_access.model.*;
 import lombok.extern.slf4j.Slf4j;
```
Comment (au2) [inline service-data-access/src/test/java/gov/va/vro/abd_data_access/service/FhirClientTest.java:8]: Do not use *. That is one of the checks happen when you add spotless.

File: service-data-access/src/test/resources/test.json [hunk 2]
```diff
@@ -0,0 +1,239 @@
+{
```
Comment (au2) [inline service-data-access/src/test/resources/test.json:1]: Can you give this a better name? test.json is too generic. maybe suggests what it is or what it is testing. test-bundle-01?

**Review by au2 — CHANGES_REQUESTED** [review]
Also please update with develop.

File: service-data-access/src/test/java/gov/va/vro/abd_data_access/service/FhirClientTest.java [hunk 3]
```diff
@@ -4,10 +4,7 @@
 import ca.uhn.fhir.parser.IParser;
 import ca.uhn.fhir.rest.api.EncodingEnum;
 import gov.va.vro.abd_data_access.exception.AbdException;
-import gov.va.vro.abd_data_access.model.AbdBloodPressure;
-import gov.va.vro.abd_data_access.model.AbdClaim;
-import gov.va.vro.abd_data_access.model.AbdEvidence;
-import gov.va.vro.abd_data_access.model.AbdMedication;
+import gov.va.vro.abd_data_access.model.*;
```
[the remark this answers is not in the document]
Comment (wlin2019) [reply service-data-access/src/test/java/gov/va/vro/abd_data_access/service/FhirClientTest.java:7]: Adjusted imports and removed *.

File: service-data-access/src/test/java/gov/va/vro/abd_data_access/service/FhirClientTest.java (hunk 1 above)
Comment (wlin2019) [reply service-data-access/src/test/java/gov/va/vro/abd_data_access/service/FhirClientTest.java:13]: Removed *.

File: service-data-access/src/test/resources/test.json (hunk 2 above)
Comment (wlin2019) [reply service-data-access/src/test/resources/test.json]: Renamed.

**Review by au2 — APPROVED** [review]

**Review by ianwolson — APPROVED** [review]

**Pull request merged**